store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2022 Northern.tech AS
+//
+//    All Rights Reserved
+
+package mongo
+
+import (
+	"context"
+
+	"github.com/mendersoftware/deviceauth/model"
+	"github.com/mendersoftware/go-lib-micro/log"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+	mstore_v1 "github.com/mendersoftware/go-lib-micro/store"
+	mstore "github.com/mendersoftware/go-lib-micro/store/v2"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	findBatchSize = 100
+)
+
+type migration_2_0_0 struct {
+	ds  *DataStoreMongo
+	ctx context.Context
+}
+
+// Up creates an index on status and id in the devices collection
+func (m *migration_2_0_0) Up(from migrate.Version) error {
+	_false := false
+	_true := true
+	var expireAfterSeconds int32 = 0
+	tokenExpirationIndexName := "TokenExpiration"
+	currentDbName := mstore_v1.DbFromContext(m.ctx, DbName)
+	logger := log.FromContext(m.ctx)
+
+	collectionsIndexes := map[string]struct {
+		Indexes []mongo.IndexModel
+	}{
+		DbDevicesColl: {
+			Indexes: []mongo.IndexModel{
+				// create device index on status and id
+				{
+					Keys: bson.D{
+						{Key: model.DevKeyId, Value: 1},
+						{Key: model.TenantIDField, Value: 1},
+					},
+					Options: &mopts.IndexOptions{
+						Background: &_false,
+						Name:       &indexDevices_Status,
+						Unique:     &_false,
+					},
```

Comment:
The options have a cleaner functional interface :)
Also, the `Background` option is deprecated since mongo 4.2.
```suggestion
					Options: mopts.Index().
						SetName(indexDevices_Status).
						SetUnique(false),
```

Reply:
Thanks! Refactored